enterprise/server/backends/authdb/authdb.go

```diff
@@ -296,10 +372,22 @@ func (d *AuthDB) GetAPIKeyGroupFromAPIKey(ctx context.Context, apiKey string) (i
 		}
 		return nil, err
 	}
+	if d.apiKeyGroupCache != nil {
+		metrics.APIKeyLookupCount.With(prometheus.Labels{metrics.APIKeyLookupStatus: "cache_miss"}).Inc()
+		d.apiKeyGroupCache.Add(apiKey, akg)
+	} else {
+		metrics.APIKeyLookupCount.With(prometheus.Labels{metrics.APIKeyLookupStatus: "invalid_key"}).Inc()
```

Comment:
it this in the right place? It seems like it will fire any time apiKeyGroupCache is nil.

Reply:
woops, this was also broken in the old code